torch/lib/TensorLib/Utils.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "TensorLib/CPUGenerator.h"
+
+namespace tlib {
+
+#define TLIB_ASSERT(cond, ...) if (!cond) { tlib::runtime_error(__VA_ARGS__); }
+
+[[noreturn]]
+void runtime_error(const char *format, ...);
+
+template <typename T, typename Base>
+static inline T* checked_cast(Base* expr, const char * name, int pos) {
+  if(auto result = dynamic_cast<T*>(expr))
```

Comment:
We can eventually get rid of this. Type already has enough meta-data in it to do it, but Storage and Generators also get passed through this interface.